examples/empirical_frechet_mean_uncertainty_s3.py

```diff
@@ -0,0 +1,184 @@
+"""
+Uncertainty of the empirical Fréchet mean on the sphere
+
+The variance of the Fréchet mean FM_n of a sample of n IID
+random variables of variance Var is decreasing more slowly
+in a sphere than in a Euclidean space.
+This example computes the  modulation factor
+     alpha = Var( FM_n) / ( n * Var)
+for isotropic distributions on hyperspheres or radius 0<r<Pi
+in the sphere s3.
+"""
+
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import geomstats.backend as gs
+# import geomstats.visualization as visualization
+
+from geomstats.geometry.hypersphere import Hypersphere
+
+# n = number of samples for computing the mean
+# NN = number of trials for the stochastic integral of expectation
+# Ntheta = number of points on the curve sig_est(theta)
+
+def my_random_uniform_sphere(n):
+    #locals(x,xn)
+    # Random coordinates between -1 and 1
+    x = (np.random.random_sample(n+1)*2.0 -1.0)
+    nx = np.linalg.norm(x)
+    while nx > 1 or nx < 1.e-10:
+       x = (np.random.random_sample(n+1)*2.0 -1.0)
+       nx = np.linalg.norm(x)
+    return(x/nx)
+
+def my_random_sample(N, dim):
```

Comment:
Thanks for the beginning of the vectorized rewriting. I am not sure to know how to complete the while loop so that the redraw is limited to the draw_again indices (the good samples should not be redrawn otherwise they could become bad the next time). But in any case the function was removed.